mne/io/neuralynx/neuralynx.py

```diff
@@ -98,31 +139,133 @@ def __init__(
         # the sample sizes of all segments
         n_segments = nlx_reader.header["nb_segment"][0]
         block_id = 0  # assumes there's only one block of recording
-        n_total_samples = sum(
-            nlx_reader.get_signal_size(block_id, segment)
-            for segment in range(n_segments)
+
+        # get segment start/stop times
+        start_times = np.array(
+            [nlx_reader.segment_t_start(block_id, i) for i in range(n_segments)]
+        )
+        stop_times = np.array(
+            [nlx_reader.segment_t_stop(block_id, i) for i in range(n_segments)]
         )
 
-        # construct an array of shape (n_total_samples,) indicating
-        # segment membership for each sample
-        sample2segment = np.concatenate(
-            [
-                np.full(shape=(nlx_reader.get_signal_size(block_id, i),), fill_value=i)
-                for i in range(n_segments)
+        # find discontinuous boundaries (of length n-1)
+        next_start_times = start_times[1::]
+        previous_stop_times = stop_times[:-1]
+        seg_diffs = next_start_times - previous_stop_times
+
+        # mark as discontinuous any two segments that have
+        # start/stop delta larger than sampling period (1/sampling_rate)
+        logger.info("Checking for temporal discontinuities in Neo data segments.")
+        delta = 1 / info["sfreq"]
```

Comment:
Just to be careful about float arithmetic being inexact (if you were using `int` sample numbers you would be safe, but time is float) it might be better to put the boundary at the midpoint of the sample numbers (one sample or two samples later, i.e., 1.5 samples later) like:
```suggestion
        delta = 1.5 / info["sfreq"]
```

Reply:
Thanks! OK, so if I follow correctly it might be safest to:
1. compute start/stop temporal differences (`float`)
2. express those time differences in `int` samples (e.g. `seg_diffs * sampling_rate)`
3. redefine delta in terms of `int` samples (i.e. not `float` sampling period)
4. and check if any `int(gaps) > int(delta)` (in which case the above correction of delta from `1/sr` to `1.5/sr` is not needed, since we'd be doing exact int ops?)

Happy to do either way, just double-checking on my end!